### PR TITLE
Translate 'turtle' adventure to German

### PR DESCRIPTION
**Pull request by DerSibi**
**Description**

Provides a German translation for the 'turtle' adventure.

**Fix for**

Describe what issue or discussion your pr concerns

**How to test**

Describe how to run your code to see it is action. See this https://github.com/Felienne/hedy/pull/880#issue-1016304308 for an example.

**Checklist**

If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [ ] Links to an existing issue or discussion (if not, create an issue first)
- [ ] Describes changes clear in the format above (present tense, no subject)
- [ ] Has a "how to test" section
- [ ] Only one thing is done in this pull request (specifically please try to refrain from mixing textual changes to the yamls from code changes)

### CODE REVIEW

**Comment by DerSibi** [comment]
Hey @das-g,

still, I seem to be able to invite only selected people to PRs - and you seem not to be one of them :( (Your name won't show up when searching for it).

Anyway, would you like to have a look?

Sebastian

**Comment by das-g** [comment]
> Hey @das-g,
> 
> still, I seem to be able to invite only selected people to PRs - and you seem not to be one of them :( (Your name won't show up when searching for it).

That's perfectly normal. As far as GitHub is concerned, this repo is a personal repo of Felienne. So for requested reviews, her name is available as a reviewer. (And maybe the names of others to whom she has granted write/push permissions on the repo.) I, like you, am so far just another "outside" contributor to Hedy without write/push permissions, so GitHub doesn't let you request reviews from me through their "Request Review" functionality. (I and every other GitHub user able to comment on PRs here can still _give_ reviews. But you'll note that when we approve a PR, it'll only give a grey check-mark not a green one. This is to indicate that our reviews/approvals aren't made in the name of the project or of the project owner. These reviews are just outside contributions, much like our PRs themselves.)

> Anyway, would you like to have a look?

Sure!

**Comment by Felienne** [comment]
I invited both of you to the repo, so I think that should allow you to invite each other to PR's

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:696]: - `random` is a Hedy keyword, but is it a command ("Befehl")?
- In the English version, the "with" is essential in communicating that these things can be combined and used _together_. German "während" just indicates that they happen concurrently. Can we convey this better somehow?

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:697]: If we want it closer to the English version:
```suggestion
                    Durch eine zufällige Wahl nimmt die Schildkröte jedes Mal einen anderen Weg.
```
(Though your version is fine, too.)

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:702]:
```suggestion
                    ## Hedy-Beispiel-Code
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:718]:
```suggestion
                    In Level 4 musst Du bei `print` und `ask` Anführungszeichen nutzen.
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:722]:
```suggestion
                    ## Hedy-Beispiel-Code
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:681]: As this is printed out, I think it's more like a title or an announcement / status report to the user, rather than a command for the turtle. Thus maybe:
```suggestion
                    print Formen zeichnen
```
(Of course, "Zeichne formen" could not just be imperative, but also a short form of "Ich (Computer / Programm / Schildkröte) zeichne formen.", so it could be used an announcement / status report, too.)

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:725]: I think "Formen" captures the meaning better than "Figuren". Also, a colon probably doesn't make much sense when no further textual terminal output follows.

**Comment by DerSibi** [comment]
@das-g, thanks a lot for the explanation. Let's see if it works now :)

**Comment by DerSibi** [comment]
> I invited both of you to the repo, so I think that should allow you to invite each other to PR's

Doesn't work :(

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:750]:
```suggestion
                    Form is ask 'Möchtest Du ein Quadrat oder ein Dreieck?'
```
The user doesn't have to draw. We made them a program that draws for them! :tada:

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:764]:
```suggestion
                    Form is ask 'Möchtest Du ein Quadrat oder ein Dreieck?'
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:775]:
```suggestion
                    ## Hedy-Beispiel-Code
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:772]:
```suggestion
                    Wir setzen den Winkel einmalig und nutzen dann die Variable im `repeat`.
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:784]:
```suggestion
                    ## Interaktion
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:787]:
```suggestion
                    Vervollständige den Code und Du kannst jedes Vieleck zeichnen, das Du möchtest.
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:764]: Note to self: Should maybe be something like `number_of_corners` rather than `angles` in English?

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:795]: As this isn't a list, maybe
```suggestion
                    Eckenanzahl = ask 'Wie viele Ecken soll die Form haben?'
                    Winkel = 360 / Eckenanzahl
                    repeat Eckenanzahl times
```
or
```suggestion
                    Anzahl_Ecken = ask 'Wie viele Ecken soll die Form haben?'
                    Winkel = 360 / Anzahl_Ecken
                    repeat Anzahl_Ecken times
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:714]: It's unfortunate that "Winkel" is the same in plural and singular, so the `angle` vs. `angles` distinction cannot be made so easily in German. Should we maybe deviate from standard German grammar and make it `Winkels`, as in Haskell and some German dialects / some variants of colloquial German? Or should we use a more complex variable name like `WinkelAuswahl`?

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:804]:
```suggestion
                    In diesem Level kannst Du mit dem Befehl `repeat` eine Code-Zeile wiederholen.
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:807]:
```suggestion
                    ## Hedy-Beispiel-Code
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:820]:
```suggestion
                    In diesem Level nutzen wir Berechnungen, um verschiedene Formen zu zeichnen.
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:827]:
```suggestion
                    Jetzt, da wir mit Hedy Mathe machen können, können wir alle Formen zeichnen, die wir wollen!
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:830]:
```suggestion
                    ## Hedy-Beispiel-Code
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:834]: again, should we put "Anzahl" ("number of …") into the variable name for the number of corners?

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:1135]: Keep that space, so that the echoed choice won't be glued to the colon!
```suggestion
                    echo so your choice was: 
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:1456]: Let's not regress w.r.t. #1848
```suggestion
                     print Thank you for your order!
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:1455]: Keep that space, so that the echoed ordered thing(s) won't be glued to the word "order"!
```suggestion
                     echo So you would like to order 
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:1838]:
```suggestion
                    echo Your name is 
```

File: coursedata/adventures/de.yaml
Comment (das-g) [inline coursedata/adventures/de.yaml:2099]:
```suggestion
                    echo I choose door 
```

**Review by das-g — CHANGES_REQUESTED** [review]

**Comment by Felienne** [comment]
> > I invited both of you to the repo, so I think that should allow you to invite each other to PR's
> 
> Doesn't work :( When I click on the gear icon and try to type in das-g, he won't show up in the list.

I assumed that was before he accepted the invite, I suspect it should work now!

**Comment by das-g** [comment]
> I assumed that was before he accepted the invite, I suspect it should work now!

Sorry, I've only now (ca. 19:45 MET, 18:45 UTC) accepted the invitation.

File: coursedata/adventures/de.yaml
Comment (DerSibi) [reply coursedata/adventures/de.yaml]: Oh dear... What was I thinking..?

File: coursedata/adventures/de.yaml
Comment (DerSibi) [reply coursedata/adventures/de.yaml]: Uh, nice catch!

File: coursedata/adventures/de.yaml
Comment (DerSibi) [reply coursedata/adventures/de.yaml:834]: See above. Wouldn't complicate things.

File: coursedata/adventures/de.yaml
Comment (DerSibi) [reply coursedata/adventures/de.yaml:1135]: Will be replaced later on anyway though. Before starting the next adventure, I always copy the remainder of the English file. So, it is fixed there already, I'll copy it over soon anyway.

File: coursedata/adventures/de.yaml
Comment (DerSibi) [reply coursedata/adventures/de.yaml:1455]: See above

File: coursedata/adventures/de.yaml
Comment (DerSibi) [reply coursedata/adventures/de.yaml:1456]: See above.

File: coursedata/adventures/de.yaml
Comment (DerSibi) [reply coursedata/adventures/de.yaml:1838]: See above

**Review by DerSibi — COMMENTED** [review]
Thanks a lot @das-g

File: coursedata/adventures/de.yaml
Comment (das-g) [reply coursedata/adventures/de.yaml:772]: With "im `repeat`" I tried to mirror the (similarly imprecise) "in the `repeat`" of the English version, understanding it as a way to say "in the `repeat` block`" without having to spell out the (indeed somewhat technical) "block". Neither are ideal, but for now, I'd go with "im".

File: coursedata/adventures/de.yaml
Comment (das-g) [reply coursedata/adventures/de.yaml:784]: > You insist on it?

Not at all. I don't see a reason to not translate the English literally here (which would be the noun "Interaktion", not the adjective "interaktiv"), but as a title of this section, both work comparably well.

File: coursedata/adventures/de.yaml
Comment (das-g) [reply coursedata/adventures/de.yaml:787]: > Can you live with a bit of deviating from the original?

Sure. But I'm unsure whether the current deviation doesn't change the meaning too much.

"[Z]eichne jedes Vieleck, das Du möchtest." can be understood a task — and a rather daunting one at that. To keep it almost as short, how about

```suggestion
                    Vervollständige den Code und er zeichnet jedes Vieleck, das Du möchtest.
```

File: coursedata/adventures/de.yaml
Comment (das-g) [reply coursedata/adventures/de.yaml:795]: > Ecken is the plural of Ecke.

Yes, and I appreciate that you used two different words ("Ecke" vs. "Winkel"), so children can distinguish when we mean "corner" (part of a shape) and when "angle" (a property/measure of a corner).

However, I think it can be confusing when we sometimes use plural variable names for variables that hold several things/values of said sort or type, and other times use plural variable names to hold a single number that indicates the number of such things. But as the English original doesn't make that distinction, we should maybe discuss it in general and/or for English first and until a decision is reached there stick to a more or less verbatim translation (i.e., keep it as-is without mentioning "Anzahl" in the variable names, even if the variable holds a number.)

File: coursedata/adventures/de.yaml
Comment (das-g) [reply coursedata/adventures/de.yaml:1135]: So, it'd be broken from when this translation here (turtle adventure) will be merged until the adventure after that will be translated _and_ merged? If so, is there any reason to simply _not_ remove the space here, so that merging this PR won't break this in the first place? I don't think we should knowingly introduce regressions to the `main` branch and the production website, if we can easily avoid it.

**Comment by Felienne** [comment]
Thanks for this @DerSibi!! So nice to have more German content available!

File: coursedata/adventures/de.yaml
Comment (DerSibi) [reply coursedata/adventures/de.yaml:1135]: On second thought, I think you're totally right. It was rather about not spoiling the commit history (and maybe getting into trouble with merges later). However, in the English language file, it's written without the space. Correcting it now here makes no sense to me, given that I want to copy the original English text over before starting a new translation. And still, I totally dislike introducing an error here. I am uncertain how to proceed.

File: coursedata/adventures/de.yaml
Comment (DerSibi) [reply coursedata/adventures/de.yaml:1456]: Same here, actually... It looks like the regression has taken place already. In addition, this block is ill-formatted. The code block is indented one space too far. I'll hopefully keep it on the list for when I reach it.

File: coursedata/adventures/de.yaml
Comment (DerSibi) [reply coursedata/adventures/de.yaml:2099]: Same here. I suggest filing an issue to correct it in the EN language file. I'd suppose that it then finds its way into other language files as well.

File: coursedata/adventures/de.yaml
Comment (Felienne) [reply coursedata/adventures/de.yaml:1135]: Hi @DerSibi! Simply removing the space will fix the issue as I understand it? If you want to you can commit the fix from the browser:

image

I can also do it for you if you want. And don't worry all too much about introducing errors, all code in adventures will be tested before we deploy it.

You can see here that all the tests are ok for this translation:

image

Comment (DerSibi) [reply coursedata/adventures/de.yaml:1135]: Hi @Felienne, 
thanks for hooking in. It's rather a general question. @das-g and me agree, that there should be a white space in general. However, the EN file does not contain a white space there. And I would like to keep the English parts in the DE-file (that are yet to be translated to German) identical to the EN-file. Even if there are small errors in it. The reason for this is all and foremost, that I follow your advice to copy remaining English parts into the DE file from the EN file before translating a new adventure to have the latest changes.

So, fixing the white space in the DE file now doesn't make sense to me, because I'd overwrite it again before starting the next adventure - or having to fix it manually over and over again. Also, there's a handful of further places in this file below this line with similar issues. So, what's your opinion: Fix the whitespace error or leave it that way?

Comment (das-g) [reply coursedata/adventures/de.yaml:1135]: Sorry, I wasn't aware the spaces were now also missing in the English version, and thus misunderstood you, @DerSibi: I thought you were arguing that copying the English file would add the spaces back in later, so we wouldn't have to do so in this PR. Now I understand that you were making us aware that adding them back in within this PR would be somewhat moot, as copying the English file would remove them again later.

We definitely should fix that in the English file. (Or should the behavior of the `echo` command be modified to output a space automatically between its argument and the echoed input, at least in Hedy Levels that don't require string quoting? Requiring trailing spaces in Hedy code for producing the usually expected output might be confusing, as one can't see them in Hedy's code input field.)

Comment (DerSibi) [reply coursedata/adventures/de.yaml:1135]: +1 regarding urge to fix it in EN file. But for this PR, I'd resolve this point.

File: coursedata/adventures/de.yaml
Comment (das-g) [reply coursedata/adventures/de.yaml]: I meant to let it refer to "den Code", thus "er". Assigning agency to the turtle with "sie" would work, too, as it has been mentioned in the sentence before this one.

File: coursedata/adventures/de.yaml
Comment (das-g) [reply coursedata/adventures/de.yaml:1456]: > It looks like the regression has taken place already.

Indeed. Probably happened when resolving merge conflicts in bfe8bcb40ccbebb5d3a81f35a5059b1384afce47 within #1838.

Comment (Felienne) [reply coursedata/adventures/de.yaml:1456]: > Let's not regress w.r.t. #1848

This change is in main, so it will be fixed automatically when we update this branch from main. No need to keep this PR up to date with all others.

**Comment by Felienne** [comment]
Hi @DerSibi, I looked into the space issue and it is a relic from a time when `echo` worked a bit differently, it all works without space now:

image

So once you translate this adventure to German, you can also skip the space.

Thanks again fro this hard work, it will be available in a few minutes on https://hedy-alpha.herokuapp.com/ if you want to check it in action, and will be release to the site when we deploy next (often in the weekend)

**Review by Felienne — APPROVED** [review]

**Comment by DerSibi** [comment]
Hi @Felienne, 

> So once you translate this adventure to German, you can also skip the space.

ah, good to know. Thanks for the hint.

> Thanks again fro this hard work, it will be available in a few minutes on https://hedy-alpha.herokuapp.com/ if you want to check it in action, and will be release to the site when we deploy next (often in the weekend)

My pleasure. I'll start with one of the next adventures soon, I guess. Thanks for telling me about this staging site. I'll pay it a visit :)

**Pull request merged**
